### PR TITLE
[RNMobile] Ungrouping for Group blocks

### PR DESCRIPTION
**Pull request by jbinda**
## Description
Add `Ungroup` icon on the` BlockControls` toolbar when `Group` block is selected and it's possible to ungroup the inner blocks. After click it removes the `Group` block holding the `InnerBlocks` components which are moved one level up in the hierarchy.

PR is connected with [#1347](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1347) in gutenberg-mobile.

Please also refer to:
[Related gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1381)

## How has this been tested?
Manual test on Android and iOS plus CI tests

## Screenshots 
image

## Types of changes
New Feature with Ungroup Button

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by jbinda** [comment]
@pinarol @iamthomasbishop what do you think about that approach ? 

I have used already implemented logic to ungroup, just add mobile implementation for `UngroupButton` and add some logic under `Group` block

**Comment by iamthomasbishop** [comment]
I think that should work. The icon isn't super obvious IMO, but it should work for now.

Semi-related note: we really need to add the selected block icon/indicator to the toolbar asap – this is a great example of how confusing it could be with only the ungroup icon and no indication that you've got a group block selected.

**Comment by jbinda** [comment]
@iamthomasbishop 

Great! I will manage CI to pass and @Tug review as well.

Also I agree that it is confusing for the user about current selection depending only on icon. However @dratwas already raised PR with breadcrumbs and I did dimming and dash-boarder feature under Navigation down PR. I believe it will be merged soon after we deal with Floating Toolbar feature (@lukewalczak is going to update the PR with our last fixes on making it clickable without problems)

Isn't it enough to inform user about the selection ?

File: packages/block-library/src/group/edit.native.js [hunk 1]
```diff
@@ -29,9 +31,12 @@ function GroupEdit( {
 	}
 
 	return (
-		<InnerBlocks
-			renderAppender={ isSelected && InnerBlocks.ButtonBlockAppender }
-		/>
+		<>
+			<UngroupButton />
```
Comment (Tug) [inline packages/block-library/src/group/edit.native.js]: I don't think it's good practice to import a component from `@wordpress/editor` in the block library.

The editor should depend on the block editor which should depend on the block library not the other way around. In general I think we should try to stay as close as possible to the web structure.

In that case they seem to be using the `__experimentalBlockSettingsMenuPluginsExtension` component from `@wordpress/block-editor` to inject the `ConvertToGroupButton` into the `BlockSettingsMenu`.
If that's too much overhead for such a simple PR, we can simply have the `Ungroup` button in the `@wordpress/block-editor` for now and add it directly to the `BlockToolbar`.

**Review by Tug — CHANGES_REQUESTED** [review]
I had a comment about the code structure.
I think we could add that part as well to our `BlockToolbar` component:
https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-toolbar/index.js#L20
We don't need to implement `BlockSettingsMenu` for now if we don't want to though, we can have `Ungroup` here instead.

```
if ( blockClientIds.length > 1 ) {
	return (
		<Ungroup />
	);
}
```
Edit: Actually that's for grouping, not ungrouping, so we don't need that for now

File: packages/block-library/src/group/edit.native.js (hunk 1 above)
Comment (jbinda) [reply packages/block-library/src/group/edit.native.js]: Ok - thank you for the advice. I have moved src file of button under `@wordpress/block-editor`

However I need this button only for group block so if its ok to render this button as it is currently ?

File: packages/block-editor/src/components/ungroup-button/index.native.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * Internal dependencies
+ */
+import { Ungroup } from './icon';
+import { BlockControls } from '../block-controls';
+
+export function UngroupButton( {
+	onConvertFromGroup,
+	isUngroupable = false,
+} ) {
+	return ( isUngroupable &&
```
Comment (lukewalczak) [inline packages/block-editor/src/components/ungroup-button/index.native.js]: Rewrite it to `if-else` statement.

File: packages/block-library/src/group/edit.native.js (hunk 1 above)
Comment (Tug) [reply packages/block-library/src/group/edit.native.js]: Still, I'd rather we remove that import from `@wordpress/block-editor` inside this block component.
You should be able to test if it's a group component from the BlockToolbar instead

Comment (jbinda) [reply packages/block-library/src/group/edit.native.js]: Ok, WDYT now ?

Comment (Tug) [reply packages/block-library/src/group/edit.native.js]: That looks better thank you :)

File: packages/block-library/src/group/edit.native.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ import {
 	InnerBlocks,
 	withColors,
 } from '@wordpress/block-editor';
+
```
Comment (Tug) [inline packages/block-library/src/group/edit.native.js]: You probably want to revert this change as well

File: packages/block-editor/src/components/ungroup-button/index.native.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	ToolbarButton,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { Ungroup } from './icon';
```
Comment (Tug) [inline packages/block-editor/src/components/ungroup-button/index.native.js]: Could we name it `UngroupIcon` or UngroupSVG`?

**Review by Tug — COMMENTED** [review]
Haven't product reviewed yet but code looks good beside the few comments I had 👍

File: packages/block-editor/src/components/ungroup-button/index.native.js (hunk 4 above)
Comment (jbinda) [reply packages/block-editor/src/components/ungroup-button/index.native.js]: Sure!

File: packages/block-library/src/group/edit.native.js (hunk 3 above)
Comment (jbinda) [reply packages/block-library/src/group/edit.native.js]: Youre rigth

File: packages/block-editor/src/components/ungroup-button/index.native.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		const isUngroupable = selectedBlock && selectedBlock.innerBlocks && !! selectedBlock.innerBlocks.length;
+		const innerBlocks = isUngroupable ? selectedBlock.innerBlocks : [];
+
+		return {
+			isUngroupable,
+			clientId: selectedId,
+			innerBlocks,
+		};
+	} ),
+	withDispatch( ( dispatch, { clientId, innerBlocks, onToggle = noop } ) => {
```
Comment (Tug) [inline packages/block-editor/src/components/ungroup-button/index.native.js]: One more thing, I'm curious what this onToggle is used for?

**Comment by iamthomasbishop** [comment]
> Also I agree that it is confusing for the user about current selection depending only on icon. However @dratwas already raised PR with breadcrumbs and I did dimming and dash-boarder feature under Navigation down PR.

@jbinda That's right, I forgot about the breadcrumbs showing the selected block icon. That should be fine for now, we can tackle the task of adding to the Quick Toolbar separately. 👍

File: packages/block-editor/src/components/ungroup-button/index.native.js (hunk 4 above)
Comment (jbinda) [reply packages/block-editor/src/components/ungroup-button/index.native.js]: done

File: packages/block-library/src/group/edit.native.js (hunk 3 above)
Comment (jbinda) [reply packages/block-library/src/group/edit.native.js]: done

File: packages/block-editor/src/components/ungroup-button/index.native.js (hunk 5 above)
Comment (jbinda) [reply packages/block-editor/src/components/ungroup-button/index.native.js]: This button implementation comes from the web version and I believe that this is gateway to pass some action after ungroup.

I'm not sure how it is use on web and at this moment I didn't noticed any reason that we may want to pass callback on mobile version. I left it there for consistence with web but perhaps if your going to say that if we do not use it we should remove it I would agree...

**Comment by jbinda** [comment]
hey @tug
I will change target of this branch to `master` tomorrow morning. Will you be able to take a quick look once again if everything's fine ? How about this `Toggle()` method. Would you like to remove it ?

**Comment by jbinda** [comment]
I have retarget to `master` and it's ready to merge now.

**Review by Tug — APPROVED** [review]
Code LGTM 👍
